sunpy/net/dataretriever/sources/goes.py

```diff
@@ -168,9 +178,95 @@ def _can_handle_query(cls, *query):
         boolean
             answer as to whether client can service the query
         """
-        chkattr = ['Time', 'Instrument', 'SatelliteNumber']
+        chkattr = ["Time", "Instrument", "SatelliteNumber"]
         chklist = [x.__class__.__name__ in chkattr for x in query]
         for x in query:
-            if x.__class__.__name__ == 'Instrument' and x.value.lower() in ('xrs', 'goes'):
+            if x.__class__.__name__ == "Instrument" and x.value.lower() in (
+                "xrs",
+                "goes",
+            ):
                 return all(chklist)
         return False
+
+
+class SUVIClient(GenericClient):
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        Returns a URL to the SUVI data for the specified date.
+
+        Parameters
+        ----------
+        timerange: sunpy.time.TimeRange
+            time range for which data is to be downloaded.
+
+        level : str
+            The level of the data.
+
+        Note
+        ----
+        SUVI data does not exist for GOES satellite < 16.
+        """
+        base_url = "https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes{goes_number}/"
+
+        if "wavelength" not in kwargs.keys() or not kwargs["wavelength"]:
+            raise ValueError("Queries must specify a wavelength.")
+        else:
+            wavelength = kwargs["wavelength"]
+        wavelength = wavelength.to(u.Angstrom, equivalencies=u.spectral())
+
+        if "satellitenumber" in kwargs.keys():
+            satellitenumber = int(kwargs["satellitenumber"])
+        else:
+            satellitenumber = 16
+
+        if "level" in kwargs.keys():
+            level = str(kwargs["level"])
+        else:
+            level = "2"
+
+        if str(level) == "2":
+            search_pattern = base_url + 'l{level}/data/suvi-l{level}-ci{wave:03}/%Y/%m/%d/dr_suvi-l{level}-ci{wave:03}_g16_s%Y%m%dT%H%M%SZ_.*\.fits'
+        elif str(level) == "1b":
+            if wavelength in u.Quantity([131, 171, 195, 284], 'Angstrom'):
+                search_pattern = base_url + 'l{level}/suvi-l{level}-fe{wave:03}/%Y/%m/%d/OR_SUVI-L{level}-Fe{wave:03}_G16_s%Y%j%H%M%S.*\.fits.gz'
+            elif wavelength == 304 * u.Angstrom:
+                search_pattern = base_url + 'l{level}/suvi-l{level}-he{wave:03}/%Y/%m/%d/OR_SUVI-L{level}-He{wave_minus1:03}_G16_s%Y%j%H%M%S.*\.fits.gz'
+            elif wavelength == 94 * u.Angstrom:
+                search_pattern = base_url + 'l{level}/suvi-l{level}-fe{wave:03}/%Y/%m/%d/OR_SUVI-L{level}-Fe{wave_minus1:03}_G16_s%Y%j%H%M%S.*\.fits.gz'
+        else:
+            return []
```

Comment:
I think you want to modify `_can_handle_query` to not be a valid query for this client if it's level 1b and not one of the allowed wavelengths?